tracing/span.go

```diff
@@ -366,35 +342,138 @@ func (s *Span) createChild(name string, spanType SpanType, component string) *Sp
 //     clientCtx, span := parentSpan.ChildAndThriftContext(ctx, "myCall")
 //     result, err := client.MyCall(clientCtx, arg1, arg2)
 //     span.End(ctx, err)
-func (s *Span) ChildAndThriftContext(ctx context.Context, name string) (context.Context, *Span) {
-	ctx, span := s.CreateClientChildForContext(ctx, name)
-	ctx = CreateThriftContextFromSpan(ctx, span)
-	return ctx, span
+func ChildAndThriftContext(ctx context.Context, span Span, name string) (context.Context, ClientSpan) {
+	ctx, child := CreateClientChildForContext(ctx, span, name)
+	ctx = CreateThriftContextFromSpan(ctx, child)
+	return ctx, child
 }
 
-// SetServerSpan sets the span as the ServerSpan on the given context
-func (s *Span) SetServerSpan(ctx context.Context) (context.Context, error) {
-	if s.spanType != SpanTypeServer {
-		return nil, &InvalidSpanTypeError{SpanTypeServer, s.spanType}
+type contextKey int
+
+const (
+	serverSpanKey contextKey = iota
+	childSpanKey
+)
+
+// FlagMask values.
+//
+// Reference: https://github.com/reddit/baseplate.py/blob/1ca8488bcd42c8786e6a3db35b2a99517fd07a99/baseplate/observers/tracing.py#L60-L64
+const (
+	// When set, this trace passes all samplers.
+	FlagMaskDebug int64 = 1
+)
+
+const (
+	counterKeyPrefix   = "counter."
+	baseplateComponent = "baseplate"
+)
+
+type baseSpan struct {
+	trace *Trace
+	hooks *SpanHooks
```

Comment:
nit: this does not need to be a pointer (all slices are actually pointers anyway).

Reply:
Probably not, I can also do the simpler `type` suggestion.